allure-support/src/main/kotlin/com/kaspersky/components/alluresupport/AllureSupportKaspressoBuilder.kt

```diff
@@ -48,10 +46,81 @@ fun Kaspresso.Builder.addAllureSupport(): Kaspresso.Builder = apply {
+    if (!isAndroidRuntime) {
+        return@simple
+    }
+    customize.invoke(this)
+    val instrumentalDependencyProvider =
+        instrumentalDependencyProviderFactory.getComponentProvider<Kaspresso>(instrumentation)
+    forceAllureSupportFileProviders(instrumentalDependencyProvider)
+    addRunListenersIfNeeded(instrumentalDependencyProvider)
+}.apply(::addAllureSupportInterceptors)
+
+private fun Kaspresso.Builder.forceAllureSupportFileProviders(provider: InstrumentalDependencyProvider) {
+    resourcesDirNameProvider = DefaultResourcesDirNameProvider()
+    resourceFileNamesProvider = DefaultResourceFileNamesProvider(addTimestamps = false)
+
+    val allureResourcesRootDirsProvider = DefaultAllureResourcesRootDirsProvider()
+    resourcesRootDirsProvider = allureResourcesRootDirsProvider
+
+    val defaultDirsProvider = DefaultDirsProvider(provider)
+    val allureDirsProvider = AllureDirsProvider(defaultDirsProvider, instrumentation, allureResourcesRootDirsProvider)
+    dirsProvider = allureDirsProvider
+
+    val defaultResourcesDirsProvider = DefaultResourcesDirsProvider(allureDirsProvider, resourcesDirNameProvider)
+    resourcesDirsProvider = defaultResourcesDirsProvider
+
+    val defaultResourceFilesProvider =
+        DefaultResourceFilesProvider(allureResourcesRootDirsProvider, defaultResourcesDirsProvider, resourceFileNamesProvider)
+    val allureResourcesFilesProvider =
+        AllureResourceFilesProvider(defaultResourceFilesProvider, allureResourcesRootDirsProvider, resourcesDirsProvider)
+    resourceFilesProvider = allureResourcesFilesProvider
+}
+
+private fun Kaspresso.Builder.addRunListenersIfNeeded(provider: InstrumentalDependencyProvider) {
+    provider.runNotifier.apply {
+        addUniqueListener(::AllureRunListener)
+        addUniqueListener {
+            AllureResultsHack(
+                uiDevice = provider.uiDevice,
+                resourcesRootDirsProvider = resourcesRootDirsProvider as DefaultAllureResourcesRootDirsProvider,
+                dirsProvider = dirsProvider as AllureDirsProvider
+            )
+        }
+    }
+}
+
+private fun addAllureSupportInterceptors(builder: Kaspresso.Builder): Unit = with(builder) {
+    if (!isAndroidRuntime) {
+        return@with
+    }
+    val provider = instrumentalDependencyProviderFactory.getComponentProvider<Kaspresso>(instrumentation)
+    val allureResourcesFilesProvider = resourceFilesProvider as AllureResourceFilesProvider
+    stepWatcherInterceptors.addAll(
+        listOf(
+            ScreenshotStepInterceptor(screenshots),
+            AllureMapperStepInterceptor()
+        )
+    )
+    testRunWatcherInterceptors.addAll(
+        listOf(
+            DumpLogcatTestInterceptor(logcatDumper),
+            ScreenshotTestInterceptor(screenshots),
+            DumpViewsTestInterceptor(viewHierarchyDumper),
+            HackyVideoRecordingTestInterceptor(videos, allureResourcesFilesProvider, provider.runNotifier.getUniqueListener())
```

Comment:
Taking the hack out of the listeners list. Another option is to have `AllureResultsHack` instance simply in static. It would be simpler and not so much different in behavior, as the instance is single in both ways and dies with the process.